mlrun/feature_store/retrieval/base.py

```diff
@@ -198,34 +198,58 @@ def _generate_vector(
                     self._append_drop_column(column)
                     column_names.append(column)
 
+            if isinstance(timestamp_for_filtering, typing.Dict):
+                time_column = (
+                    timestamp_for_filtering.get(name, None)
+                    or feature_set.spec.timestamp_key
+                )
+            elif isinstance(timestamp_for_filtering, str):
+                time_column = timestamp_for_filtering
+            else:
+                time_column = feature_set.spec.timestamp_key
+
+            if (start_time or end_time) and (
+                not time_column
+                or (
+                    time_column != feature_set.spec.timestamp_key
+                    and time_column
+                    not in [feature.name for feature in feature_set.spec.features]
+                )
+            ):
+                logger.info(
+                    f"You provided start_time or end_time but the feature_set {name} "
+                    f"doesn't have timestamp_key or {timestamp_for_filtering} you spesified for this feature set "
+                    f"is not in it's features therefore the time filter didn't executed on it."
```

Comment:
Need to split this into two separate checks and errors. Then, please run the texts through a spell checker.